src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -127,10 +127,8 @@ public function normalize($data, $format = null, array $context = array())
         if (null === $data || is_scalar($data)) {
             return $data;
         }
-        if (is_object($data) && $this->supportsNormalization($data, $format)) {
-            return $this->normalizeObject($data, $format, $context);
-        }
-        if ($data instanceof \Traversable) {
+
+        if (is_array($data) || $data instanceof \Traversable) {
```

Comment:
This is not good. A custom normalizer can be able to handle `\Traversable` instances and must but be called first in such case.

For instance see: https://github.com/dunglas/DunglasApiBundle/blob/1.x/Hydra/Serializer/CollectionNormalizer.php#L71-L74

It would be nice to add a test to cover this case.